addons/stock/models/ir_actions_report.py

```diff
@@ -14,3 +19,62 @@ def _get_rendering_context(self, report, docids, data):
                 'docs': docs,
             })
         return data
+
+    def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
```

Comment:
Why do you need that? It's often not a good practice to copy/paste a full function. You will end up with a lot of duplicated code and people will probably never patch the function twice when needed. Instead, try to find a sub function to hook your code. If it's not possible, then we can discuss finding another solution

Reply:
I agree. There was another solution for this that I have implemented in this [commit](https://github.com/odoo/odoo/pull/145546/commits/afe9871c477a51c583094602e11e531a8d576080). This commit modifies the base function to accept multiple records with the same ID by just replacing the `dict` with a `list`. I didn't like this solution because modifying the base may result in more issues in other places. That's why I ended up with inheriting the base `IrActionsReport`.